src/registry.js

```diff
@@ -88,12 +88,21 @@ class Config {
     this._data.components = components;
   }
 
-  resolve(component) {
-    return this._data.components[component] || {};
+  getComponent(name, options = {}) {
+    const { context = '' } = options;
+    const componentName = `${name}${context ? `|${context}` : ''}`;
```

Comment:
I went for a string based composition for name... We could go for the key based one as well.